src/index.js

```diff
@@ -1,118 +1,149 @@
       }
-    : name(props || {}, children)
 }
 
+/**
+ * Start an application.
+ * @param {Object} state The state tree of the application.
+ * @param {Object} actions The actions tree of the application.
+ * @param {Function} view A function that returns a virtual node.
+ * @param container A DOM element to render the application to.
+ */
 export function app(state, actions, view, container) {
-  var patchLock
-  var lifecycle = []
-  var root = container && container.children[0]
-  var node = vnode(root, [].map)
-
-  repaint(init([], (state = copy(state)), (actions = copy(actions))))
+  /**
+   * Prevents unnecessary view renders when calling multiple actions in a row.
+   **/
+  var renderLock
+
+  /** Stores lifecycle event callbacks that will be invoked after patching the DOM. */
+  var lifecycleStack = []
+  /**
+   * The first element child of the container.
+   **/
+  var root = (container && container.children[0]) || null
+  /**
+   * The last virtual node computed from the view.
+   **/
+  var node = root && elementToVNode(root, [].map)
+
+  // Copy the state and actions tree to avoid mutating non-owned objects.
+
+  scheduleRender(
+    wireStateToActions([], (state = copy(state)), (actions = copy(actions)))
+  )
 
   return actions
 
-  function vnode(element, map) {
-    return (
-      element && {
-        name: element.nodeName.toLowerCase(),
-        props: {},
-        children: map.call(element.childNodes, function(element) {
-          return element.nodeType === 3
-            ? element.nodeValue
-            : vnode(element, map)
-        })
-      }
-    )
+  /** Create a virtual node from a DOM tree. */
+  function elementToVNode(element, map) {
+    return {
+      name: element.nodeName.toLowerCase(),
+      props: {},
+      children: map.call(element.childNodes, function(element) {
+        return element.nodeType === 3
+          ? element.nodeValue
+          : elementToVNode(element, map)
+      })
+    }
   }
 
+  /** Compute the next virtual node. */
   function render(next) {
-    patchLock = !patchLock
+    renderLock = !renderLock
     next = view(state, actions)
 
-    if (container && !patchLock) {
+    if (container && !renderLock) {
       root = patch(container, root, node, (node = next))
     }
 
-    while ((next = lifecycle.pop())) next()
+    while ((next = lifecycleStack.pop())) next()
   }
 
-  function repaint() {
-    if (!patchLock) {
-      patchLock = !patchLock
+  function scheduleRender() {
+    if (!renderLock) {
+      renderLock = !renderLock
       setTimeout(render)
     }
   }
 
-  function copy(a, b) {
-    var target = {}
+  function copy(target, source) {
+    var obj = {}
 
-    for (var i in a) target[i] = a[i]
-    for (var i in b) target[i] = b[i]
+    for (var i in target) obj[i] = target[i]
+    for (var i in source) obj[i] = source[i]
 
-    return target
+    return obj
   }
 
-  function set(path, value, source, target) {
+  /** Set the value at path of the source. */
```

Comment:
Not sure we need this comment (same for `get`) since the functions reads like 80% of what the comment says haha